lxd/container_lxc.go

```diff
@@ -1634,9 +1634,13 @@ func (c *containerLXC) initLXC(config bool) error {
 				}
 			}
 
-			err = lxcSetConfigItem(cc, fmt.Sprintf("%s.%d.script.up", networkKeyPrefix, networkidx), fmt.Sprintf("%s callhook %s %d network-up %s", c.state.OS.ExecPath, shared.VarPath(""), c.id, k))
-			if err != nil {
-				return err
+			// Check if the container has network specific keys set to avoid unnecessarily running the network up hook.
+			containerNetworkKeys := append(containerNetworkLimitKeys, containerNetworkRouteKeys...)
+			if shared.StringMapHasStringKey(m, containerNetworkKeys...) && shared.StringInSlice(m["nictype"], []string{"bridged", "p2p"}) {
```

Comment:
Can you split this into its own earlier commit so we can cherry-pick it into stable?

That is, the earlier commit shouldn't mention containerNetworkRouteKeys, then this commit should just alter the logic to also cover containerNetworkRouteKeys.

Reply:
@stgraber this is now done